core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/LanguageInPropertyShape.java

```diff
@@ -38,6 +40,13 @@
 		super(id, connection, nodeShape, deactivated, parent, path);
 
 		this.languageIn = toList(connection, languageIn).stream().map(Value::stringValue).collect(Collectors.toSet());
+		Iterator value = this.languageIn.iterator();
+		String x = "( ";
+		while (value.hasNext()) {
+			x += "'" + value.next() + "' ";
+		}
+		x += ")";
```

Comment:
Can you pull this out in a separate method. Something like langeuageInForExpected();